packages/pug-lexer/index.js

```diff
@@ -971,6 +971,31 @@ Lexer.prototype = {
     }
   },
 
+  /**
+   * EachOf.
+   */
+
+  eachOf: function() {
+    var captures;
+    if (captures = /^(?:each|for) (.*) * of *([^\n]+)/.exec(this.input)) {
+      this.consume(captures[0].length);
+      var tok = this.tok('eachOf', captures[1]);
+      tok.value = captures[1] || null;
```

Comment:
This should never be `null`.

```suggestion
      tok.value = captures[1];
```

Reply:
I think we should have something like:

```js
if (!(/^[a-zA-Z_$][\w$]*$/.test(tok.value.trim()) || /^\[ *[a-zA-Z_$][\w$]* *\, *[a-zA-Z_$][\w$]* *\]$/.test(tok.value.trim())) {
      this.error(
        'MALFORMED_EACH_OF_LVAL',
        'The value variable for each must either be a valid identifier (e.g. `item`) or a pair of identifiers in square brackets (e.g. `[key, value]`).'
      );
}
```

We can always update this to be more flexible later, if we are willing to actually validate that it's a valid JS destructuring pattern, but allowing everything now makes it difficult to safely add validation later.